### DISCUSSION TITLE
Open large dataset without loading the whole file in RAM

### DISCUSSION THREAD

**Discussion by rsatapat** [pydata/xarray#8113] [Q&A] [answered]
### What is your issue?

Is it possible to load a large dataset (larger than available RAM) by providing preconditions similar to standard indexing (for example, sel(dim_x = value))? If not, is it possible to add this functionality?

I am planning to use xarray to store data gathered from neuroscience experiments that tend to by quite large and might come across this issue.

**Comment by TomNicholas** [comment, marked as answer]
Xarray opens files lazily by default, and our lazy classes understand indexing, so what you're asking for should already work by default. If you want to actually process larger-than-RAM datasets (not just view subsections of them) you will need to install [dask](https://docs.xarray.dev/en/stable/user-guide/dask.html).

**Reply by TomNicholas** [reply]
@dirtywind please let us know if you run into problems trying this!

**Reply by rsatapat** [reply]
All good! I wasn't aware of lazy computing and dask since I am relatively new to coding but, I got it working just as I wanted it.
Thanks! :)
